Rivet/Functions/Checkpoint-Migration.ps1

```diff
@@ -66,14 +74,18 @@ function Checkpoint-Migration
         }
 
         Write-Debug "Checkpoint-Migration: Exporting migration on database $($databaseItem.Name)"
-        $migration = Export-Migration -SqlServerName $settings.SqlServerName -Database $databaseItem.Name
-        $migration = $migration -join [Environment]::NewLine
-        Set-Content -Path $OutputPath -Value $migration
+        $exportedMigration = Export-Migration -SqlServerName $settings.SqlServerName -Database $databaseItem.Name
+        $exportedMigration = $exportedMigration -join [Environment]::NewLine
 
-        foreach( $migration in $pushedMigrations )
+        # Now pop those pushed migrations and remove their respective migration files.
+        foreach( $pushedMigration in $pushedMigrations )
         {
-            $migrationFilePath = Join-Path -Path $databaseItem.MigrationsRoot -ChildPath "$($migration.MigrationFileName).ps1"
+            Invoke-Rivet -Pop -Database $databaseItem.Name -Environment $Environment -ConfigFilePath $ConfigFilePath -Name $pushedMigration.MigrationFileName

```

Comment:
Don't pop any existing migrations. Their changes will be removed from the database and then won't get captured in the schema export.

Reply:
The export happens on line 78 before the Pop happens.

Reply:
Regardless, don't pop any migrations. Just export the current schema and delete all applied migrations.